data/transactions/logic/box.go

```diff
@@ -45,20 +45,45 @@ func (cx *EvalContext) availableBox(name string, operation BoxOperation, createS
 	}
 
 	dirty, ok := cx.available.boxes[basics.BoxRef{App: cx.appID, Name: name}]
+
+	newAppAccess := false
+	// maybe allow it (and account for it) if a newly created app is accessing a
+	// box. we allow this because we know the box is empty upon first touch, so
+	// we don't have to go to the disk. but we only allow one such access for
+	// each spare (empty) box ref. that way, we can't end up needing to write
+	// many separate newly created boxes.
+	if !ok && cx.Proto.EnableUnnamedBoxAccessInNewApps {
+		if _, newAppAccess = cx.available.createdApps[cx.appID]; newAppAccess {
+			if cx.available.unnamedAccess > 0 {
```

Comment:
Is there anyway one could have this be say 3 (for 3 unnamed box refs) and the same new boxref then gets touched 3 times?

Reply:
Once the new box uses up one of these "tickets" it doesn't use up more when it is touched again.  And indeed those later touches are sure not to go to disk, because the box in question will now be in the StateDelta.